### DISCUSSION TITLE
2021/07/how-to-automate-aws-cloudfront-distribution-folder-changes-when-deploying-a-static-web-site/

### DISCUSSION THREAD

**Discussion by giscus** [SeanKilleen/seankilleen.github.io#744] [Specific Post Comments]
# 2021/07/how-to-automate-aws-cloudfront-distribution-folder-changes-when-deploying-a-static-web-site/

I was working on a project that deployed its UI, an Angular site, to an S3 bucket that was then placed behind CloudFront.

https://seankilleen.com/2021/07/how-to-automate-aws-cloudfront-distribution-folder-changes-when-deploying-a-static-web-site/

**Comment by peter-fosseus-silicon** [comment]
Thank you, you saved me hours of work. Here is an AzureDevOps pipeline piece if someone else is searching for a way to implement Sean's excellent work, I used $(Build.BuildId) as the folder.

variables:

  awsCredentialsDev: 'dev_deploy-from-azure'
  awsCodeBucket: 'dev-code-bucket'
  awsRegionName: 'af-south-1'
  awsDistributionID: 'E2P6PBCZ12348'
  awsOriginID: "dev-admin"

  - task: S3Upload@1
    inputs:
      awsCredentials: '$(awsCredentialsDev)'
      regionName: '$(awsRegionName)'
      bucketName: '$(awsCodeBucket)'
      sourceFolder: 'dist/Portal/'
      globExpressions: '**'
      targetFolder: '$(Build.BuildId)'
      createBucket: true
    displayName: 'Upload objects to S3'

  - task: AWSShellScript@1
    inputs:
      awsCredentials: '$(awsCredentialsDev)'
      regionName: '$(awsRegionName)'
      scriptType: 'inline'
      inlineScript: |
        echo "---- Start CF update ----"
        aws cloudfront get-distribution-config --id $(awsDistributionID) > output.json
        echo "----- ORIGINAL JSON -----"
        cat output.json
        echo "----- RUNNING JQ -----"
        cat output.json | jq "del(.ETag) | (.DistributionConfig.Origins.Items[] | select(.Id == \"$(awsOriginID)\")).OriginPath = \"/$(Build.BuildId)\" | .DistributionConfig" > updated-config.json
        echo "----- UPDATED JSON: -----"
        cat updated-config.json
        echo "----- GETTING ETAG VALUE FOR LATER USE -----"
        etag=`cat output.json | jq -r ".ETag"`
        echo $etag
        echo "----- CALLING TO UPDATE -----":
        aws cloudfront update-distribution --id $(awsDistributionID) --if-match="$etag" --distribution-config file://updated-config.json
        aws cloudfront wait distribution-deployed --id $(awsDistributionID) 
        aws cloudfront create-invalidation --distribution-id $(awsDistributionID) --paths "/*"
